### PR TITLE
Allocates a 3D array

### PR DESCRIPTION
**Pull request by ShawnToubeau**
3D arrays for days babyyyyy

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -1,6 +1,22 @@
 #include <stdio.h>
 #include <stdlib.h>
-#include "mpi.h"
+#include <time.h>
+// #include "mpi.h"
+
+// Calculates the conflict scores of a chess board
+int computeConflictScores(int N, int board[N][N])
```
Comment (Ethan-Arrowood) [inline main.c]: Finish this method

File: main.c [hunk 2]
```diff
@@ -9,20 +25,64 @@ int main(int argc, char **argv)
   int rank, totalRanks, num_sets, N;
   double gen_div_factor;
   // Initialize MPI
-  MPI_Init(&argc, &argv);
+  // MPI_Init(&argc, &argv);
```
Comment (Ethan-Arrowood) [inline main.c]: Add back MPI

**Review by Ethan-Arrowood — CHANGES_REQUESTED** [review]
Looks great!

**Comment by Ethan-Arrowood** [comment]
Looking good!

**Review by Ethan-Arrowood — APPROVED** [review]

File: main.c [hunk 3]
```diff
@@ -1,6 +1,31 @@
+#include "mpi.h"
+#include "assistFunctions.h"
+#include "assistFunctions.c"
```
Comment (Ethan-Arrowood) [inline main.c]: shouldn't have to include the `.c` if you have the header file

File: assistFunctions.c [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+// Calculates the conflict scores of a chess board
+int computeConflictScore(int N, int board[N])
+{
+    int i = 0, j = 1; // Double iterators
+    int(*conflictQueens) = malloc(sizeof(int[N]));
+
+    // Set all values in conflict queen to -1
+    for (; i < N * N; i++)
```
Comment (Ethan-Arrowood) [inline assistFunctions.c]: why until i is `N * N` when conflictQueens is only of size `N`?

File: assistFunctions.c [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    {
+        conflictQueens[i] = -1;
+    }
+
+    while (i != N * N)
```
Comment (Ethan-Arrowood) [inline assistFunctions.c]: I believe `i` would still be `N * N` at this point due to the previous for loop

File: assistFunctions.c [hunk 6]
```diff
@@ -0,0 +1,76 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include "omp.h"
+
```
Comment (Ethan-Arrowood) [inline assistFunctions.c:5]: missing `#include "assistFunctions.h"`

Comment (ShawnToubeau) [reply assistFunctions.c:5]: Done

File: assistFunctions.c (hunk 4 above)
Comment (ShawnToubeau) [reply assistFunctions.c]: N is only 1 dimension of the array, so if N = 8, our board would really be N * N = 64

Comment (ShawnToubeau) [reply assistFunctions.c]: Also I fudged the allocation of that array, it's fixed now 😄

File: main.c (hunk 3 above)
Comment (ShawnToubeau) [reply main.c]: Done!

File: assistFunctions.c (hunk 5 above)
Comment (ShawnToubeau) [reply assistFunctions.c]: And fixed!

**Review by Ethan-Arrowood — APPROVED** [review]
Looks good enough to merge

**Pull request merged**